go-controller/pkg/cni/helper_linux.go

```diff
@@ -365,17 +365,41 @@ func ConfigureOVS(ctx context.Context, namespace, podName, hostIfaceName string,
 	// Find and remove any existing OVS port with this iface-id. Pods can
 	// have multiple sandboxes if some are waiting for garbage collection,
 	// but only the latest one should have the iface-id set.
-	uuids, _ := ovsFind("Interface", "_uuid", "external-ids:iface-id="+ifaceID)
-	for _, uuid := range uuids {
-		if out, err := ovsExec("remove", "Interface", uuid, "external-ids", "iface-id"); err != nil {
-			klog.Warningf("Failed to clear stale OVS port %q iface-id %q: %v\n  %q", uuid, ifaceID, err, out)
+	names, _ := ovsFind("Interface", "name", "external-ids:iface-id="+ifaceID)
+	for _, name := range names {
+		if name == hostIfaceName {
+			// this may be result of restarting ovnkube-node, and it is trying to add the same VF representor to
+			// br-int for the same pod; do not delete port in this case.
+			continue
+		}
+		if out, err := ovsExec("--if-exists", "--with-iface", "del-port", "br-int", name); err != nil {
```

Comment:
Do we need `if-exists` here when we have found the interfaces above? I would rather error out if it is unexpected that the interface was removed while iterating through all the interfaces.

Reply:
I disagree. if it does not exist, it is not an error we care.

Reply:
Not even for debugging (such as a warning)? Because wouldn't be strange if the interface doesn't exist while we got the list of interfaces above?

Reply:
I personally think we won't do anything about it if we saw this warning, as it is harmless. But sure, I can remove ""--if-exists",  I don't have strong opions against it.